Add null check for drawingArea in RemoteScrollingCoordinatorProxy::layerTreeHost()

nmoucht opened the pull request:
#### b712db6f6c9516e6148cfac7c990e511f40daa42
<pre>
Add null check for drawingArea in RemoteScrollingCoordinatorProxy::layerTreeHost()
<a href="https://bugs.webkit.org/show_bug.cgi?id=242207">https://bugs.webkit.org/show_bug.cgi?id=242207</a>
&lt;rdar://85499764&gt;

Reviewed by Simon Fraser.

Add null check for fix of rdar://85499764.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.cpp:
(WebKit::RemoteScrollingCoordinatorProxy::layerTreeHost const):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.mm:
(WebKit::ScrollingTreeScrollingNodeDelegateIOS::findActingScrollParent):

Canonical link: <a href="https://commits.webkit.org/252028@main">https://commits.webkit.org/252028@main</a>
</pre>

smfr commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.cpp:
```diff
@@ -67,6 +67,10 @@ ScrollingNodeID RemoteScrollingCoordinatorProxy::rootScrollingNodeID() const
+    if (!drawingArea)
+        return nullptr;
+    
     if (!is<RemoteLayerTreeDrawingAreaProxy>(drawingArea)) {
```
`is<>` on a pointer type already does a null check so you don't have to change anything in this function.

smfr replied:
Which leaves me confused because the crash stack suggested that we were de-referencing a null RemoteLayerTreeDrawingAreaProxy

nmoucht replied:
I believe Matt hit the nail on the head that the compiler optimized this away because we were unconditionally dereferencing the result in ScrollingTreeScrollingNodeDelegateIOS::findActingScrollParent(). Adding the null check for the host there should be enough to have the compiler not optimize away the null check in RemoteScrollingCoordinatorProxy::layerTreeHost().

smfr reviewed (approved):
Don't forget to put the radar number into the commit message.

nmoucht committed:
Add null check for drawingArea in RemoteScrollingCoordinatorProxy::layerTreeHost()
https://bugs.webkit.org/show_bug.cgi?id=242207
<rdar://85499764>

Reviewed by Simon Fraser.

Add null check for fix of rdar://85499764.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.cpp:
(WebKit::RemoteScrollingCoordinatorProxy::layerTreeHost const):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.mm:
(WebKit::ScrollingTreeScrollingNodeDelegateIOS::findActingScrollParent):

Canonical link: https://commits.webkit.org/252028@main
